src/devices/markisol.c

```diff
@@ -0,0 +1,136 @@
+    buf[2] & 0x20 - bit 0 of zone
+    buf[2] & 0xd0 - bits 0,2,3 of control
+    buf[3] & 0x10 - bit 1 of control
+    buf[3] & 0x80 - bit 1 of zone
+    buf[3] & 0x6f - unknown; for my remotes (buf[3] & 0x6f) == 0x01 always
+*/
+
+#include "decoder.h"
+
+static int markisol_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t buf[5];
+    uint8_t cksum = 0;
+    int got_proper_row_length = 0;
+    for (int i = 0; i < bitbuffer->num_rows; i++) {
+        if (decoder->verbose > 0)
+            fprintf(stderr, "%s: bits_per_row[%d] = %d\n", __func__, i, bitbuffer->bits_per_row[i]);
+        if (bitbuffer->bits_per_row[i] == 41 || bitbuffer->bits_per_row[i] == 42) {
+            uint8_t *b = bitbuffer->bb[i];
+            for (int j = 0; j < 5; ++j) {
+                buf[j] = ~reverse8(((b[j] << 8) + b[j+1]) >> 7);
+                cksum += buf[j];
+            }
+            got_proper_row_length = 1;
+            break;
+        }
+    }
+
+    if (!got_proper_row_length)
+        return DECODE_ABORT_EARLY;
+
+    if (decoder->verbose > 0)
+        fprintf(stderr, "%s: %02x %02x %02x %02x %02x cksum=%02x\n", __func__, buf[0], buf[1], buf[2], buf[3], buf[4], cksum);
+
+    if (cksum != 1)
+        return DECODE_FAIL_MIC;
+
+    int address = (buf[0] << 8) | buf[1];
+    int channel = buf[2] & 0xf;
+    int control = ((buf[2] >> 4) & ~2) | ((buf[3] & 0x10) >> 3);
+    int zone = ((buf[2] & 0x20) >> 5) + ((buf[3] & 0x80) >> 6) + 1;
+    // buf[3] seems to be always 0x01, 0x11, 0x81, 0x91
+    // ... so there are 6 bits that seem constant (for my remotes)
+
+    char const *const control_strs[] = {
+            "Limit (0)", // seems like Limit=0 for channel=1, otherwise Limit=13
+            "Down (1)",
+            "? (2)",
+            "H-Down (3)",
+            "Confirm (4)",
+            "Stop (5)",
+            "? (6)",
+            "? (7)",
+            "? (8)",
+            "? (9)",
+            "? (10)",
+            "? (11)",
+            "Up (12)",
+            "Limit (13)",
+            "H-Up (14)",
+            "? (15)",
+    };
+
+    /* clang-format off */
+    data_t *data = data_make(
+            "model",          "Model",          DATA_STRING, "Markisol",
+            "id",             "",               DATA_FORMAT, "%04X", DATA_INT, address,
+            "control",        "Control",        DATA_STRING, control_strs[control],
+            "channel",        "Channel",        DATA_INT,    channel,
+            "zone",           "Zone",           DATA_INT,    zone,
+            NULL);
+    /* clang-format on */
+
+    decoder_output_data(decoder, data);
+    return 1;
+}
+
+static char *output_fields[] = {
+        "model",
+        "id",
+        "control",
+        "channel",
+        "zone",
+        NULL,
+};
+
+// rtl_433 -f 433900000 -X 'n=name,m=OOK_PWM,s=368,l=704,r=10000,g=10000,t=0,y=5628'
+
+r_device markisol = {
+        .name           = "Markisol", // E-Motion, BOFU, BF-30x, BF-415, etc.
```

Comment:
This can be a list of supported devices, e.g. "Markisol, E-Motion, BOFU, BF-30x, BF-415 curtain remote"

Reply:
Done.  Also added "Rollerhouse".